astacus/coordinator/plugins/clickhouse/steps.py

```diff
@@ -151,6 +153,30 @@ async def run_step(self, cluster: Cluster, context: StepsContext) -> Sequence[Ac
         return access_entities
 
 
+@dataclasses.dataclass
+class RetrieveUserDefinedFunctionsStep(Step[Sequence[UserDefinedFunction]]):
+    zookeeper_client: ZooKeeperClient
+    replicated_user_defined_zookeeper_path: str | None
+
+    async def run_step(self, cluster: Cluster, context: StepsContext) -> Sequence[UserDefinedFunction]:
+        if self.replicated_user_defined_zookeeper_path is None:
+            return []
+
+        user_defined_functions: list[UserDefinedFunction] = []
+        async with self.zookeeper_client.connect() as connection:
+            change_watch = ChangeWatch()
+            # In case the node does not exist, create it now.
+            await connection.try_create(self.replicated_user_defined_zookeeper_path, b"")
```

Comment:
I don't think a Retrieve function should create something, it should only read the system under backup.